src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -165,19 +167,28 @@ func onBootstrapControlPlane(s *state.State, initConfig map[string]string) error
 	if nodeIP == nil {
 		return fmt.Errorf("failed to parse node IP address %q", s.Address().Hostname())
 	}
-	certificates := pki.NewControlPlanePKI(pki.ControlPlanePKIOpts{
-		Hostname:          s.Name(),
-		IPSANs:            []net.IP{nodeIP},
-		Years:             10,
-		AllowSelfSignedCA: true,
-	})
 
 	// Create directories
 	if err := setup.EnsureAllDirectories(snap); err != nil {
 		return fmt.Errorf("failed to create directories: %w", err)
 	}
 
+	// cfg.Network.ServiceCIDR may be "IPv4CIDR,IPv6CIDR".
+	// TODO: handle ip6 if we have an ip6 cidr
+	ip4ServiceCIDR := strings.Split(cfg.Network.ServiceCIDR, ",")[0]
+	ip4ServiceIP, err := utils.GetFirstIP(ip4ServiceCIDR)
+	if err != nil {
```

Comment:
This is not right. The service CIDR might be any of `IPv4CIDR`, `IPv6CIDR` or `IPv4CIDR,IPv6CIDR`. It's not right to assume that the first address is the IPv4 service CIDR, as handled here.

Have a look at https://github.com/canonical/k8s-snap/blob/b81e215d3d48427eab52e3b975a41be7e9f9f9b9/src/k8s/pkg/component/network.go#L21-L41 for a way to handle this, but we must handle it properly here, this is what the PR is doing, let's not leave as a TODO.

You could perhaps extract this as a util function like

```
func GetKubernetesServiceIPsFromServiceCIDRs(serviceCIDR string) ([]net.IP, error)
```

where you return a list of 1 or 2 IPs from the given cidrs